buildengine/build.go

```diff
@@ -3,43 +3,63 @@ package buildengine
 import (
 	"context"
 	"fmt"
+	"path/filepath"
+	"strings"
 
 	"github.com/TBD54566975/ftl/backend/schema"
-	"github.com/TBD54566975/ftl/common/moduleconfig"
 	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/TBD54566975/ftl/internal/slices"
 )
 
-// A Module is a ModuleConfig with its dependencies populated.
-type Module struct {
-	moduleconfig.ModuleConfig
-	Dependencies []string
-}
-
-// LoadModule loads a module from the given directory.
-//
-// A [Module] includes the module configuration as well as its dependencies
-// extracted from source code.
-func LoadModule(ctx context.Context, dir string) (Module, error) {
-	config, err := moduleconfig.LoadModuleConfig(dir)
-	if err != nil {
-		return Module{}, err
+// Build a module in the given directory given the schema and project config.
+func Build(ctx context.Context, sch *schema.Schema, project Project) error {
+	switch project := project.(type) {
+	case Module:
+		return buildModule(ctx, sch, project)
+	case ExternalLibrary:
+		return buildExternalLibrary(ctx, sch, project)
+	default:
+		return fmt.Errorf("unsupported project type: %T", project)
 	}
-	return UpdateDependencies(ctx, config)
 }
 
-// Build a module in the given directory given the schema and module config.
-func Build(ctx context.Context, sch *schema.Schema, module Module) error {
+func buildModule(ctx context.Context, sch *schema.Schema, module Module) error {
 	logger := log.FromContext(ctx).Scope(module.Module)
 	ctx = log.ContextWithLogger(ctx, logger)
+
 	logger.Infof("Building module")
 	switch module.Language {
 	case "go":
-		return buildGo(ctx, sch, module)
-
+		return buildGoModule(ctx, sch, module)
 	case "kotlin":
-		return buildKotlin(ctx, sch, module)
-
+		return buildKotlinModule(ctx, sch, module)
 	default:
 		return fmt.Errorf("unknown language %q", module.Language)
 	}
 }
+
+func buildExternalLibrary(ctx context.Context, sch *schema.Schema, lib ExternalLibrary) error {
+	logger := log.FromContext(ctx).Scope(filepath.Base(lib.Dir))
+	ctx = log.ContextWithLogger(ctx, logger)
+
+	imported := slices.Map(sch.Modules, func(m *schema.Module) string {
+		return m.Name
+	})
+	logger.Infof("Generating stubs [%s] for %v", strings.Join(imported, ", "), lib)
```

Comment:
Make this Debugf - we're basically sending everything to debug or trace to avoid spamming the console during normal development